services/blockchain-indexer/shared/dataService/business/token.js

```diff
@@ -68,6 +82,49 @@ const getTokenBalances = async (params) => {
 	return tokens;
 };
 
+const getTokenTopBalances = async (params) => {
+	const response = {
+		data: {},
+		meta: {},
+	};
+	const accountBalancesTable = await getAccountBalancesTable();
+	const queryParams = {
+		tokenID: params.tokenID,
+	};
+
+	// Count total number of response before adding limit/offset
+	const totalAddressCount = await accountBalancesTable.count(queryParams);
+
+	if (params.limit) queryParams.limit = params.limit;
+	if (params.offset) queryParams.offset = params.offset;
+	queryParams.sort = params.sort ? params.sort : 'balance:desc';
```

Comment:
Please build the queryParams at once with all the necessary information. `count` method already ignores the pagination properties.

Reply:
Applied.